src/components/cc-domain-management/cc-domain-management.js

```diff
@@ -0,0 +1,819 @@
+
+          <cc-block-section>
+            <div class="form-info">
+              <p>${i18n('cc-domain-management.form.info.docs')}</p>
+              <p>${i18n('cc-domain-management.form.info.primary')}</p>
+              <p>${i18n('cc-domain-management.form.info.cleverapps')}</p>
+            </div>
+            <div>
+              ${this._renderForm(this.domainFormState)}
+            </div>
+          </cc-block-section>
+
+          <cc-block-section>
+            <div slot="title">
+              ${i18n('cc-domain-management.list.heading')}
+              ${this.domainListState.type === 'loaded' ? html`
+                <cc-badge class="domain-count" circle>${this._sortedDomains.length}</cc-badge>
+              ` : ''}
+            </div>
+
+            ${this.domainListState.type === 'error' ? html`
+              <cc-notice intent="warning" message="${i18n('cc-domain-management.list.loading-error')}"></cc-notice>
+            ` : ''}
+
+            ${this.domainListState.type === 'loading' ? html`
+              <cc-loader></cc-loader>
+            ` : ''}
+
+            ${this.domainListState.type === 'loaded'
+              ? this._renderDomains(this._sortedDomains)
+              : ''}
+          </cc-block-section>
+
+        </cc-block>
+
+        <cc-block>
+          <div slot="title">${i18n('cc-domain-management.certif.heading')}</div>
+          <div class="certif-info">
+            <p>${i18n('cc-domain-management.certif.automated')}</p>
+            <p>${i18n('cc-domain-management.certif.custom')}</p>
+          </div>
+        </cc-block>
+
+        <cc-block class="dns-info">
+          <div slot="title">${i18n('cc-domain-management.dns.heading')}</div>
+          <div>
+            <div class="dns-info__desc">${i18n('cc-domain-management.dns.desc')}</div>
+            <cc-notice intent="info" heading="${i18n('cc-domain-management.dns.info.heading')}">
+              <div slot="message">
+                <p>${i18n('cc-domain-management.dns.info.load-balancer')}</p>
+                <p>${i18n('cc-domain-management.dns.info.apex')}</p>
+              </div>
+            </cc-notice>
+          </div>
+
+          ${this.dnsInfoState.type === 'loading' ? html`
+            <cc-loader></cc-loader>
+          ` : ''}
+
+          ${this.dnsInfoState.type === 'error' ? html`
+            <cc-notice intent="warning" message="${i18n('cc-domain-management.dns.loading-error')}"></cc-notice>
+          ` : ''}
+
+          ${this.dnsInfoState.type === 'loaded'
+            ? this._renderDnsInfo(this.dnsInfoState.cnameRecord, this.dnsInfoState.aRecords)
+            : ''}
+        </cc-block>
+      </div>
+    `;
+  }
+
+  /**
+   * @param {DomainManagementFormState} formState
+   * @returns {TemplateResult}
+   */
+  _renderForm ({ type, hostname: domain, pathPrefix }) {
+    const isDisabled = type === 'adding';
+
+    return html`
+      <form novalidate @submit=${this._onDomainSubmit}>
```

Comment:
question: you don't use our directive. Did we miss something when we designed the form framework? Or do you plan to use it in a next iteration?

Reply:
Yes this is temporary, the directive landed in master right when the component was going into review so I've created an issue about this and it's probably one of the first subjects I'll tackle after shipping the component